### PR TITLE
Do not rebuild series index on delete for series not overlapping in time

### PR DESCRIPTION
**Pull request by hpbieker**
Do not rebuild series index on delete for series not overlapping in time. See #10504.

###### Required for all non-trivial PRs
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by hpbieker** [comment]
@jwilder Would it be possible to get this merged into master and 1.7?

File: tsdb/engine/tsm1/engine.go [hunk 1]
```diff
@@ -1497,6 +1492,27 @@ func (e *Engine) deleteSeriesRange(seriesKeys [][]byte, min, max int64) error {
...
+
+	if !overlapsTimeRangeMinMax && e.Cache.store.count() > 0 {
+		overlapsTimeRangeMinMax = true
+	}
+
+	if !overlapsTimeRangeMinMax {
+		return nil
+	}
+
+	// Ensure keys are sorted since lower layers require them to be.
```
Comment (hpbieker) [inline tsdb/engine/tsm1/engine.go]: Moving the sort after we shortcut on overlapsTimeRangeMinMax

File: tsdb/engine/tsm1/engine.go [hunk 2]
```diff
@@ -1497,6 +1492,27 @@ func (e *Engine) deleteSeriesRange(seriesKeys [][]byte, min, max int64) error {
...
 
+	overlapsTimeRangeMinMax := false
+	e.FileStore.Apply(func(r TSMFile) error {
+		if r.OverlapsTimeRange(min, max) {
+			overlapsTimeRangeMinMax = true
+		}
+		return nil
+	})
+
+	if !overlapsTimeRangeMinMax && e.Cache.store.count() > 0 {
```
Comment (hpbieker) [inline tsdb/engine/tsm1/engine.go]: Ideally we could check the timerange on `e.Cache`, but checking for non-emptiness is the best I found.

**Comment by hpbieker** [comment]
@e-dard please review?

**Comment by e-dard** [comment]
@zeebo can you take a look please?

**Comment by e-dard** [comment]
Thanks for the hard work @hpbieker. Sorry it took us so long to get the time to review this.

**Comment by hpbieker** [comment]
Thank you @e-dard . How do we get this into master / 2.0 as well? I guess 2.0 is already branched off?

#10502 is also a related change (but with a different issue.

**Pull request merged**